objects/src/accounts/delta/mod.rs

```diff
@@ -58,19 +55,18 @@ impl AccountDelta {
     }
 
     /// Merge another [AccountDelta] into this one.
-    pub fn merge(self, other: Self) -> Result<Self, AccountDeltaError> {
-        let nonce = match (self.nonce, other.nonce) {
+    pub fn merge(&mut self, other: Self) -> Result<(), AccountDeltaError> {
```

Comment:
Merging into self simplifies code a bit. :)

Reply:
Make sense. There is one tradeoff to consider - this would be a breaking change for the node. So, if the interface changes, we'll need a small PR on the node side to make sure things still work.